private/buf/cmd/buf/command/push/push_test.go

```diff
@@ -146,6 +147,46 @@ func TestPush(t *testing.T) {
 	)
 }
 
+func TestPushIsMinimal(t *testing.T) {
+	// Assert push only manifests the minimal amount of files needed to build
+	// the module.
```

Comment:
Nit: May be clear to say that's not pushing a _minimal_ amount of files (sounds like some of them are optional and we're deciding not to push them), but instead we're only pushing _allowed_ files, so we're actually pushing _all of the allowed_ files, and ignoring unrelated ones.

Reply:
It's not even allowed but what's needed to build the module, according to the build section in the configuration. For example, excluded directories are not sent. "Minimal" may be misleading though, so I'll give this another wording spin.

Reply:
I see you changed to "smaller". Not strictly always smaller. User could have just one proto and `buf.yaml`, with no excluded directories, and the result built module could even be bigger (includes an empty `buf.lock`?), or at least the same amount of files.

In my mind what happens is then not an "allow" either (as I suggested) but a "curation" or "validation" or "preparation", meaning "let's see which files in this dir actually make part of the module": excluding directories, ignoring unrecognized files, generating missing `buf.lock` file, etc. So may be more like "curated" or "validated" bucket?

Not blocking ofc. But I think it helps for clarity.

Reply:
We're not checking if the module is valid (e.g. buildable), so validated is misleading. "Curate" could work but that also implies correctness of some aspect, whatever it may be in the perspective of the curator.

Perhaps "subset" would be a more appropriate term. No correctness statement is being made and one subset of a set is the set itself. Minimal / smaller concepts don't come into play.